### DISCUSSION TITLE
java.lang.ClassCastException: java.math.BigInteger cannot be cast to java.lang.Long

### DISCUSSION THREAD

**Discussion by horasjey** [leokhoa/laragon#919] [Q&A]
Hallo anyone, 
know how to fix this isue?
`java.lang.ClassCastException: java.math.BigInteger cannot be cast to java.lang.Long`

i get this error in my app netbeans

thanks

**Comment by Pen-y-Fan** [comment]
I don't know Java, I can see its cast error, you must tell Java to cast from one type to another, unlike PHP which handles it automatically. Chat GPT said this:

This error occurs because **java.math.BigInteger** and **java.lang.Long** are different classes, and Java does not allow direct casting between them. To fix this, you need to convert the **BigInteger** to a **Long** properly. Here's how you can do it:

```java
BigInteger bigIntValue = new BigInteger("12345678901234567890");
Long longValue = bigIntValue.longValue();
```

This will safely convert the **BigInteger** to a **Long** without causing a **ClassCastException**.

Does this help resolve your issue?
